### ISSUE TITLE
language_probs not implemented?

### ISSUE THREAD

**Issue by bayliner** [xenova/whisper-web#55]
```
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Whisper Transcription</title>
</head>
<body>
    <h1>Upload an audio file for transcription</h1>
    <input type="file" id="audioFile" accept="audio/*">
    <button id="transcribeButton">Transcribe</button>
    <p><strong>Transcription:</strong> <span id="transcription"></span></p>
    <p><strong>Detected Language:</strong> <span id="detectedLanguage"></span></p>

    <script type="module">
        import { pipeline } from 'https://cdn.jsdelivr.net/npm/@xenova/transformers@2.3.0';

        // Whisper model load
        let transcriber = null;

        async function loadModel() {
            transcriber = await pipeline('automatic-speech-recognition', 'Xenova/whisper-small');
            console.log("Model loaded successfully!");
        }

        loadModel();

        async function transcribe() {
            const fileInput = document.getElementById('audioFile');
            const file = fileInput.files[0];

            if (!file) {
                alert('Please select an audio file.');
                return;
            }

            if (!transcriber) {
                alert('Model is still loading. Please wait.');
                return;
            }

            const arrayBuffer = await file.arrayBuffer();
            const audioContext = new AudioContext({ sampleRate: 16000 });
            const audioBuffer = await audioContext.decodeAudioData(arrayBuffer);

            let audioData = audioBuffer.getChannelData(0);

            const output = await transcriber(audioData, {
                chunk_length_s: 30,
                task: 'transcribe',
                stride_length_s: 5,
                return_language: true,  // this!
                return_timestamps: true,
            });

            console.log("Full output:", output);
            let detectedLanguage = "Unknown";

            if (output.language) {
                detectedLanguage = output.language;
            } else if (output.metadata && output.metadata.language) {
                detectedLanguage = output.metadata.language;
            } else if (output.detected_language) {
                detectedLanguage = output.detected_language;
            }

            document.getElementById('detectedLanguage').innerText = detectedLanguage;

            document.getElementById('transcription').innerText = output.text;
        }

        document.getElementById('transcribeButton').addEventListener('click', transcribe);
    </script>
</body>
</html>
```

I'm working on with xenova whisper-web.

transcribe worked well. so good.

But I want to use return_language field, but no return value at all.

Down anyone knows this issue?

I will share if I solve the issue.

Thanks.

**Comment by xenova** [comment]
Hi there 👋 This is currently not yet implemented in Transformers.js since we didn't have any requests for this... until now!

https://github.com/huggingface/transformers.js/blob/c2ab81af062d32ad46e892e7ea5c554ca14117de/src/tokenizers.js#L3551

It shouldn't be too difficult to add though. Looks like just adding the parameter to the pipeline ([here](https://github.com/huggingface/transformers.js/blob/c2ab81af062d32ad46e892e7ea5c554ca14117de/src/pipelines.js#L1890-L1892)). Maybe you or a community member would be interested in contributing this?

**Comment by bayliner** [comment]
I want to work with! I'm very interested. How can I contribute? : )

**Comment by bayliner** [comment]
Sorry, I misunderstand.

 I need 'detect_language' task..

Could I use ' languageOutput.logits;' now? 

I can load up to small model, So I can't test large-v2


`
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Whisper Transcription</title>
</head>
<body>
    <h1>Upload an audio file for transcription</h1>
    <input type="file" id="audioFile" accept="audio/*">
    <button id="transcribeButton">Transcribe</button>
    <p><strong>Transcription:</strong> <span id="transcription"></span></p>
    <p><strong>Language Probabilities:</strong> <span id="languageProbabilities"></span></p>
    <script type="module">
        import { pipeline } from 'https://cdn.jsdelivr.net/npm/@xenova/transformers@latest';
        
        let transcriber = null;
        let languageDetector = null;
        async function loadModel() {
            transcriber = await pipeline('automatic-speech-recognition', 'Xenova/whisper-tiny.en');
            languageDetector = await pipeline('automatic-speech-recognition', 'Xenova/whisper-small', {
                task: 'detect-language',  // detect-language 대신 transcribe 사용
                return_timestamps: false
            });
            console.log("Models loaded successfully!");
        }
        loadModel();
        async function transcribe() {
            const fileInput = document.getElementById('audioFile');
            const file = fileInput.files[0];
            if (!file) {
                alert('Please select an audio file.');
                return;
            }
            if (!transcriber || !languageDetector) {
                alert('Models are still loading. Please wait.');
                return;
            }
            const arrayBuffer = await file.arrayBuffer();
            const audioContext = new AudioContext({ sampleRate: 16000 });
            const audioBuffer = await audioContext.decodeAudioData(arrayBuffer);
            let audioData = audioBuffer.getChannelData(0);
            try {
                // 언어 감지 시도 (첫 30초만 사용)
                const firstChunk = audioData.slice(0, 16000 * 30); // 30초 샘플
                //const languageOutput = await languageDetector(firstChunk);
                const languageOutput = await languageDetector(firstChunk, { return_logits: true });
                console.log("Language detection output:", languageOutput);
                
                // 트랜스크립션 수행
                const output = await transcriber(audioData, {
                    chunk_length_s: 30,
                    task: 'transcribe',
                    stride_length_s: 5,
                    return_timestamps: true,
                });
                
                console.log("Full transcription output:", output);
                document.getElementById('transcription').innerText = output.text;
                               
                if (languageOutput) {
                    const logits = languageOutput.logits;\
                    if (logits) {
                        // logits를 확률로 변환 (softmax)
                        const expScores = logits.map(x => Math.exp(x));
                        const sumExpScores = expScores.reduce((a, b) => a + b, 0);
                        const probabilities = expScores.map(exp => exp / sumExpScores);
                        
                        const languages = ['en', 'zh', 'de', 'es', 'ru', 'ko', 'fr', 'ja', 'pt', 'tr', 'pl', 'ca', 'nl', 'ar', 'sv', 'it', 'id', 'hi', 'fi', 'vi', 'he', 'uk', 'el', 'ms', 'cs', 'ro', 'da', 'hu', 'ta', 'no', 'th', 'ur', 'hr', 'bg', 'lt', 'la', 'mi', 'ml', 'cy', 'sk', 'te', 'fa', 'lv', 'bn', 'sr', 'az', 'sl', 'kn', 'et', 'mk', 'br', 'eu', 'is', 'hy', 'ne', 'mn', 'bs', 'kk', 'sq', 'sw', 'gl', 'mr', 'pa', 'si', 'km', 'sn', 'yo', 'so', 'af', 'oc', 'ka', 'be', 'tg', 'sd', 'gu', 'am', 'yi', 'lo', 'uz', 'fo', 'ht', 'ps', 'tk', 'nn', 'mt', 'sa', 'lb', 'my', 'bo', 'tl', 'mg', 'as', 'tt', 'haw', 'ln', 'ha', 'ba', 'jw', 'su'];
                        
                        const languageProbs = languages.map((lang, i) => ({
                            lang: lang,
                            prob: probabilities[i]
                        }))
                        .sort((a, b) => b.prob - a.prob)
                        .slice(0, 3);  // 상위 3개만 선택
                        
                        const probStr = languageProbs
                            .map(({lang, prob}) => `${lang}: ${(prob * 100).toFixed(2)}%`)
                            .join(', ');
                        document.getElementById('languageProbabilities').innerText = probStr;
                    }
                }
            } catch (error) {
                console.error("Error during transcription:", error);
                alert('Error during transcription. Check console for details.');
            }
        }
        document.getElementById('transcribeButton').addEventListener('click', transcribe);
    </script>
</body>
</html>
`

**Comment by bayliner** [comment]
I've also test in python env (macbook)

Even tiny model display detect_language's lang_probs.


```
import torch
import whisper

model = whisper.load_model("tiny")

audio = whisper.load_audio("./sample.wav")
audio = whisper.pad_or_trim(audio) 
mel = whisper.log_mel_spectrogram(audio).to(model.device)

detected_lang, lang_probs = model.detect_language(mel)

# sort
sorted_lang_probs = sorted(lang_probs.items(), key=lambda item: item[1], reverse=True)
print(f"language: {detected_lang}\n")
for lang, prob in sorted_lang_probs:
    print(f"{lang}: {prob:.2%}")
```


Any ideas?

**Comment by bayliner** [comment]
Misunderstood issue.
